src/java/org/apache/cassandra/streaming/StreamPlan.java

```diff
@@ -70,6 +70,16 @@ public StreamPlan(StreamOperation streamOperation, int connectionsPerHost,
     /**
      * Request data in {@code keyspace} and {@code ranges} from specific node.
      *
+     * Here, we have to encode both _local_ range transientness (encoded in Replica itself, in RangesAtEndpoint)
+     * and _remote_ (source) range transientmess, which is encoded by splitting ranges into full and transient.
+     *
+     * At the other end the distinction between full and transient is ignored it just used the transient status
+     * of the Replica objects we send to determine what to send. The real reason we have this split down to
```

Comment:
"At the other end the distinction between full and transient is ignored it just used the transient status of the Replica objects we send to determine what to send"

This isn't terribly clear - perhaps

"At the other end the distinction between full and transient is ignored; it is used only to create the Replica objects that are used to determine what to send"

This does raise the question of whether we should be sending Replica objects at all?  Presumably we *do* look at the transient/full status, in some way?  Or do we only care about the Range<Token>?  In which case, perhaps we should just send that, and avoid the confusion?

Reply:
We care https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/db/streaming/CassandraStreamManager.java#L151

Reply:
We also care about it being split up into two collections https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/dht/StreamStateStore.java#L81

It's used to determine what remote nodes have already provided us.

Reply:
Thanks, I assumed as much.  We should clarify the comment to explain why we care then, as it currently reads as though the information may be redundant.